### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by adityalahiri** [sdv-dev/CTGAN#63]
* CTGAN version: 0.2.1
* Python version: 3.5
* Operating System: Linux

### Description

If I run CTGAN twice with every setting being same and fit on same dataset, while sampling, I get different points and also different losses. How to  make it consistent. I tried setting torch random seed but didn't work.
### What I Did
from ctgan import CTGANSynthesizer
from sklearn.datasets import make_blobs
X = make_blobs()
ft = X[0]

ctgan = CTGANSynthesizer()
ctgan.fit(ft,epochs=10)
s1=ctgan.sample(100)

ctgan1 = CTGANSynthesizer()
ctgan1.fit(ft,epochs=10)
s2=ctgan1.sample(100)

I want s1 and s2 to be same.

**Comment by oregonpillow** [comment]
Hi @arrayslayer ,
I'm going to give you a very general answer because I'm not as familiar as @csala . But it's my understanding that the initial seed state that the synthesiser starts from is random, which will cause the loss function to start from different places every time you initiate a new instance of the synthesizer. Furthermore, even if we used a consistent seed value the sampling results would still be different each time since we are randomly sampling from a distribution that is learned by the model.

You would need to fork and modify the code to set a consistent seed value for synthesizer and tell the sampling method to stop randomly sampling from a distribution, instead set it to sample from a consistent point along the distribution each time. However, if you do this, my guess is that it would only be capable of producing very limited numbers of unique rows before the generated rows keep repeating themselves over and over again...(?)

Perhaps there is enough of a use-case for your problem this that this could be implemented in the future. I'm guessing it wouldn't be too hard to allow users to select a pre-defined seed state prior to fitting.

**Comment by arrayslayer** [comment]
Hi @oregonpillow ,
Thanks a lot for the detailed response. I added numpy and torch seeds to the synthesizer and transformer and it did the trick. I also set a random state in the GMM model. I get reproducible results now !
However, I am not sure how it is going to impact the fact that I could be able to only sample limited number of unique rows. I'll test that out.
If you guys want that as a user feature, I can submit a PR.
Thanks again 👍

**Comment by csala** [comment]
Awesome @arrayslayer, I'm glad you figured it out.

Yes, a PR would be welcome, though I'd appreciate if you can share a few details here beforehand about how you implemented it, so we can briefly discuss if those are all the required changes or if something else should be added to it.

**Comment by arrayslayer** [comment]
@csala 
Definitely.
So, I have set the seeds of numpy and torch in the synthesizer and transformer. I set it in the __init__. 
Also, I have set the random state in GMM model in transformer. I am attaching the git diff screenshot for your reference.

![image](https://user-images.githubusercontent.com/21020094/89288654-a077a480-d673-11ea-9fa0-9dc70c57f885.png)
